app/src/main/java/de/xikolo/viewmodels/main/NavigationViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package de.xikolo.viewmodels.main
+
+import de.xikolo.R
+import de.xikolo.managers.UserManager
+import de.xikolo.models.dao.AnnouncementDao
+import de.xikolo.models.dao.UserDao
+import de.xikolo.viewmodels.base.BaseViewModel
+import de.xikolo.viewmodels.shared.AnnouncementListDelegate
+
+class NavigationViewModel : BaseViewModel() {
+
+    private val announcementListDelegate = AnnouncementListDelegate(realm)
+
+    val announcements = announcementListDelegate.announcements
+
+    var drawerSection: Int =
+        if (UserManager.isAuthorized) {
+            R.id.navigation_my_courses
+        } else {
+            R.id.navigation_all_courses
+        }
+
+    val user
+        get() = UserDao.Unmanaged.find(UserManager.userId)
```

Comment:
You could provide a `findCurrent` method to simply return the logged in user.

Reply:
Introduced `UserDao.Unmanaged.current`.
The `LiveData`-Approach does not work as it does not observe the database query but a database object resulting in app crashes when this observed database object is deleted e.g. on logout.

<details>
<summary>Realm log</summary>
2019-07-05 21:22:26.013 32351-32351/de.xikolo.moochouse.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: de.xikolo.moochouse.debug, PID: 32351
    java.lang.IllegalStateException: Accessing object of type  which has been invalidated or deleted
        at io.realm.internal.OsObject.nativeStartListening(Native Method)
        at io.realm.internal.OsObject.addListener(OsObject.java:118)
        at io.realm.ProxyState.addChangeListener(ProxyState.java:141)
        at io.realm.RealmObject.addChangeListener(RealmObject.java:499)
        at io.realm.RealmObject.addChangeListener(RealmObject.java:542)
        at io.realm.RealmObject.addChangeListener(RealmObject.java:443)
        at de.xikolo.extensions.RealmObjectLiveData.onActive(RealmExtensions.kt:14)
        at androidx.lifecycle.LiveData$ObserverWrapper.activeStateChanged(LiveData.java:418)
        at androidx.lifecycle.LiveData$LifecycleBoundObserver.onStateChanged(LiveData.java:376)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:355)
        at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:293)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:333)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:138)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:124)
        at androidx.lifecycle.ReportFragment.dispatch(ReportFragment.java:122)
        at androidx.lifecycle.ReportFragment.onStart(ReportFragment.java:82)
        at android.app.Fragment.performStart(Fragment.java:2548)
        at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1334)
        at android.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1576)
        at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1637)
        at android.app.FragmentManagerImpl.dispatchMoveToState(FragmentManager.java:3046)
        at android.app.FragmentManagerImpl.dispatchStart(FragmentManager.java:3003)
        at android.app.FragmentController.dispatchStart(FragmentController.java:193)
        at android.app.Activity.performStart(Activity.java:7192)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3137)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1945)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:7009)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:537)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
</details>

Reply:
Ok, sad.